pages/agent/v3/help/_env_get.md

```diff
@@ -0,0 +1,68 @@
+<!--
+  _____   ____    _   _  ____ _______   ______ _____ _____ _______
+ |  __ \ / __ \  | \ | |/ __ \__   __| |  ____|  __ \_   _|__   __|
+ | |  | | |  | | |  \| | |  | | | |    | |__  | |  | || |    | |
+ | |  | | |  | | | . ` | |  | | | |    |  __| | |  | || |    | |
+ | |__| | |__| | | |\  | |__| | | |    | |____| |__| || |_   | |
+ |_____/ \____/  |_| \_|\____/  |_|    |______|_____/_____|  |_|
+
+This file is auto-generated by scripts/update-agent-help.sh, please update the
+agent CLI help in https://github.com/buildkite/agent and run the generation
+script.
+
+-->
+
+### Usage
+
+`buildkite-agent env get [variables]`
+
+### Description
+Retrieves environment variables and their current values from the current job
+execution environment. 
+
+Note that this subcommand is only available from within the job executor with
+the ′job-api′ experiment enabled.
+
+Changes to the job environment only apply to the environments of subsequent
+phases of the job. However, ′env get′ can be used to inspect the changes made
+with ′env set′ and ′env unset′.
+
+Example (gets all variables in key=value format):
+
+    $ buildkite-agent env get
+ALPACA=Geronimo the Incredible
+BUILDKITE=true
+LLAMA=Kuzco
+...
```

Comment:
Is this how the formatting needs to work in the CLI? For the docs, it would be nice to be something more like:

````
To get all the variables in `key=value` format, run:

```
buildkite-agent env get
```

The results will look like the following:

```
ALPACA=Geronimo the Incredible, BUILDKITE=true, LLAMA=Kuzco
```
````

Reply:
@mbelton-buildkite yeah. this all boils down to the process that we use to export CLI docs to the docs repo being super crufty, and not really how the cli docs are meant to be interpreted.

what i might do is kick this can down the road a bit by manually editing these files (in the docs repo) to be in the right format, leaving the next person (who will most likely also be me) to clean up the mess next time the docs get autogenerated.

it's probably also worth having a convo at some point about improving this process so that this whole feedback loop is quicker and easier.

Reply:
To me, it makes sense to display the result on multiple lines because it's supposed to simulate what the output would look like on the terminal.

Whether we put the command that the user should type and the result of that command in separate sections is less clear. I think the best experience is if the display were like the screenshot below, but there was a "copy" button that copied only the command (i.e. the line that has the `$ `, but excluding the `$ `). That would require changing the script a lot, though.

See the screenshot here https://github.com/buildkite/docs/pull/1954#issuecomment-1480621935

Reply:
Yeah, I think there are a bunch of workarounds needed to make this better, so I'm happy to leave it as-is. Manually updating it runs the risk of being overwritten with a future update.